### DISCUSSION TITLE
Zen Browser is awesome ! Here some suggestions

### DISCUSSION THREAD

**Discussion by noirbizarre** [zen-browser/desktop#1794] [Ideas]
Hi community !

This is a repost from https://www.reddit.com/r/zen_browser/comments/1freunl/zen_is_awesome_here_are_a_few_suggestions/ as suggested by @n7itro.
I inserted its answers.

I just wanted to share my feedback is I think Zen Browser is awesome. It's the first time I really stick on a browser for more than a month (I have been trying new browsers every month for the past 2 years because they all try to push annoying features I don't like and most of the time, those feature I am not using are coming in the way of my daily workflow).
Zen is so great that I think this is my first spontaneous post on Reddit !

Given it is still alpha and features are still moving, I think it's the right time to suggest some if I want a chance to have them built-in.

Note that I have zero knowledge on Firefox available APIs and so on the cost or the possibility of those. Those are pure user point of view suggestions, without any kind of authority, aggressively ... just sharing hoping some might actually be some good one.

So here they are, based on my experience.

## Merge title in URL bar

One thing that has always bothered me in any browser, and which is even more true when using vertical tabs: losing the title.
Title can be useful and is something you in fact often rely on (want to go to a tab, the picker display title).
But we don't really see the title anymore:
- with vertical tabs, even more if they are hidden, you don't see the entire title
- with horizontal tabs, the title is truncated early, then you have too many tabs open to see anything except the icon.

On the other side, I don't really care about the URL once I am on a page.

So, why not make the URL bar a title bar when you are not searching or manually entering a URL. As soon as you click or CTRL+L, it goes back to the URL/search entry.

### Picker

There have been multiple iterations on the workspace picker lately. I think it's not possible to find a place and a layout that will fit everyone (some are not even using workspaces).
So why not making it available to toolbar customization like other tools and widgets. There can be the 2 versions (only the active workspace and the workspace list).
This way, everybody is happy:
- you can choose to display it or not
- you can choose whether to have all workspaces visible or only the active one
- you can choose to have both at different places

To be perfect, the header and footer of the tab bar needs to become placeholders for this kind of buttons/widgets. Bonus, it would allow anyone to put any extension provided button in the header or footer of the tab bar.

@n7itro answered:
>  You can already choose between showing the active workspace and the icon strip in the settings

### Definition

I think it has been asked multiple times, but I'll still say it: it would be nice to allow picking any emoji of the entire official Unicode set.
I don't think selecting a subset of emojis is something that should be done anywhere (not specific to Zen).
There are literally people paid to ensure emojis are able to express as many concepts as possible in a non-verbal way.
Just let people pick any from those.

In the creation modal, it would also be nice to be able to choose the container from there.

I don't want to speak for other, but to me, a workspace is always the intersection of an identity (pro, perso, parenting...) and a topic, so I'll always have to create a workspace with:
- a descriptive name
- an emoji as close as possible from its topic 
- a specific container

@n7itro answered:
> You can add custom icons to `zen.workspaces.icons` in `about:config`. This will be added to the interface in the future

## Tabs contextual menu

I think the tab contextual menu is crowded and cluttered. Lots of entries share a semantic and I always find myself to search which menu contains my action

As of today, those entries share the same intent, sending a tab or a group of tabs somewhere (I have a French locale, so I translate it in English and might not be the exact English wording) :
- Move tab
- Send tab to device
- Open tab in container
- Change tab workspace

And opening the tab in a new window is into "move tab".

Wouldn't it be better to have a single "Send tab to" menu like this ?
- Send tab to
  - workspace
    - Workspace 1
    - Workspace 2
    - ...
  - container
    - pro
    - perso
    - ...
  - device
    - ...
  - new window
  - start
  - end

It would be even better if I could send them to other already open windows.

## Tabs drag and drop

I am used to selecting all my tabs and being able to drop them outside to open a new window or drop them on an existing window to have them in this window, but it's not working with Zen.
Is it possible to enable this behavior ?

Note that the lack of drag & drop make the previous even more important, as this is the only way for me to do that.

## Overriding extensions icons

At last, it would be great to be able to override extensions icons. Zen UI is super slick, but some extensions really have either:
- way to colorful icons 
- ugly icons

It would be really great to be able to manually override icons with some more consistent with Zen UI as I don't think each extension author will accept to change its extension icon just to match the theme of Zen.

@n7itro answer:
> You can do this with `userChrome.css` 

I didn't even think about this 😅 
This is a possibility, but not an easy one, primarily intended for developers. I'll do it for myself, but the idea was more to have it built-in and user-friendly as part of "having a slick and consistent UI".

## Enable extensions in web panels

This one is just about usability. Web panes don't use extensions.
This can be problematic, especially with password managers, (my password manager and LanguageTool spellchecker are really the one I miss the more in those).

Is it possible to enable/allow extensions in those panes ? 

## Open an empty split

Having the possibility to open a new empty split like you would for a tab or a window would be perfect.
It would allow doing:
- visiting a site
- remembering that you need some info from another site
- CTRL+<whatever> to open a right split
- search or enter URL for right split

Without extra move, without using the mouse...

## PWAs

I know this is currently worked on upstream on Firefox, so I won't spend much on this, but I choose to list it anyway as this is one of the 2 features I miss from chrome-like browsers (and it's really annoying in Chrome-based PWAs that links are not opened with the default browser);

## WebSerial, WebUSB, WebHID

Same as previous, just listed there because it's a real lack, but it's not a lack from Zen but from Firefox.
Firefox team chose not to implement them sadly, so on this one I think I'll always have to open a chrome-based browser (and there are more and more apps using it, for keyboards, hardware wallets, firmware updates...)

I think that's all (It's already a very long post for a first one).

What do you think of it ?

**Comment by ProbablyM-S** [comment]
> Merge title in URL bar

I really like this idea
maybe when you hover it shows the URL otherwise it's the title, I think it would be really cool!

**Comment by tubadan33** [comment]
+1 to Drag and Drop tabs. It does work if the tabs side bar is not hidden and the tab is dropped there, but it would be very nice to be able to drop anywhere on the new window.
